### ISSUE TITLE
Collectionview WinUI recycling item count calculation problem

### ISSUE THREAD

**Issue by enkaradag** [dotnet/maui#20206]
### Description

Hi again,

For WinUI;

In .NET MAUI version 8.0.6 (last release), CollectionView was creating extremely high count of recycle items (creating 350 400 items for a 1000 set tough 15-20 was enough for its vertical size) It was continiously creating new recycle items while scrolling. So it was flickering and app was becoming unresponsive. 

It is fixed here https://github.com/dotnet/maui/pull/20079

Now testing 8.0.7-preview.9929

CV works so much better now. But it still creates 5x more recycle items for rows. For below capture; CV needs 17 recycle items but it is creating 84 rows. When collectionview size is changed or recycle items vertical size is changed CV always creates 5x more than required recycle items. This makes whole app unresponsive for afew seconds. If a complex itemtemplate is defined that lasts longer.

I tested CV for android, ios and maccatalyst. For those it creates just enough count of recycle items.

https://github.com/dotnet/maui/assets/57816471/557ac42f-4c75-4679-929f-240c5a5b9ae7

### Link to public reproduction project repository

https://github.com/enkaradag/collectionview_flicker_test2

### Version with bug

Nightly / CI build (Please specify exact version)

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

**Comment by PureWeen** [comment]
@Foda thoughts?

**Comment by Foda** [comment]
> @Foda thoughts?

Yup -- by default WinUI [caches 4x the viewport size](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.itemsstackpanel.cachelength?view=winrt-22621). We could lower this. The cache length increasing when resized is normal, but not by another 4x amount so we should investigate.

**Comment by kevinxufei** [comment]
This issue has been verified using  Visual Studio 17.11.0 (8.0.80 & 8.0.6 & 8.0.3 & 8.0.0-rc.1.9171) & 17.11.0 (7.0.101). Can repro on Windows platform.

- **8.0.7/8.0.20/8.0.40/8.0.80:** results consistent with users to record the screen
- **8.0.3&7.0.101:** output window shows the CreatedGridCount:17, then the number increases to 500+ and an exception occurs
![q11](https://github.com/user-attachments/assets/a8e03302-3335-438c-90f1-76616727738e)
- **8.0.0-rc.1.9171:**output window displays CreatedGridCount:57
![image](https://github.com/user-attachments/assets/81fea38a-6abc-4639-942c-00ec4bc1aaaa)

**Comment by Ying-6** [comment]
This issue has been verified Visual Studio 17.14 Preview 1.1(9.0.40) debug mode. Not repro this issue on Window platform.
![Image](https://github.com/user-attachments/assets/0425586c-c36c-4d13-9647-20c737dab474)

**Comment by HarishKumarSF4517** [comment]
This issue has been verified in Visual Studio 17.14.13 with MAUI Versions (9.0.100). It can be reproduced on Windows platform both in debug and release mode. And I have attached the video for reference.

https://github.com/user-attachments/assets/74178eff-d78a-454b-9957-c44781d65482
